src/strategies/rust.ts

```diff
@@ -44,36 +44,66 @@ export class Rust extends BaseStrategy {
       }),
     });
 
-    const workspaceManifest = await this.getWorkspaceManifest();
-    const manifestPaths: string[] = [];
-    let lockPath: string;
-
-    if (
-      workspaceManifest &&
-      workspaceManifest.workspace &&
-      workspaceManifest.workspace.members
-    ) {
+    const workspaceManifest = await this.getPackageManifest();
+    const versionsMap: VersionsMap = new Map();
+
+    if (workspaceManifest?.workspace?.members) {
       const members = workspaceManifest.workspace.members;
+      if (workspaceManifest.package?.name) {
+        versionsMap.set(workspaceManifest.package.name, version);
+      } else {
+        logger.warn('No workspace manifest package name found');
+      }
+
       logger.info(
         `found workspace with ${members.length} members, upgrading all`
       );
+
+      // Collect submodule names to update
+      const manifestsByPath: Map<string, GitHubFileContents> = new Map();
       for (const member of members) {
-        manifestPaths.push(`${member}/Cargo.toml`);
+        const manifestPath = `${member}/Cargo.toml`;
+        const manifestContent = await this.getContent(manifestPath);
```

Comment:
Feels like there's enough going on here that it might benefit from being broken out into a private helper function.